### DISCUSSION TITLE
The cfg in every suit_test.go should  be as a global variable.

### DISCUSSION THREAD

**Discussion by Aisuko** [kubernetes-sigs/kubebuilder#2479] [Support]
In my view, maybe the maintainer think that the behavior is ok. So, I'm not sure open this issue as a type of bug.

As we can see that variable `cfg` not as a global variable due to the code renew `:=` it. So, the global cfg always `nil`.

We can easily test it.

```
// --omission--
Expect(cfg).ShouldNot(BeNil())


• Failure [0.000 seconds]
Test the K8S client logic
/Users/tifa/Downloads/workspace/recycleIps/pkg/ips/ips_test.go:9
  The behavior of K8S client
  /Users/tifa/Downloads/workspace/recycleIps/pkg/ips/ips_test.go:14
    Need to Init succeed [It]
    /Users/tifa/Downloads/workspace/recycleIps/pkg/ips/ips_test.go:15

    Expected
        <*rest.Config | 0x0>: nil
    not to be nil

    /Users/tifa/Downloads/workspace/recycleIps/pkg/ips/ips_test.go:29
------------------------------
STEP: Invoking the init function directly
STEP: Getting the kubernetes config from the cluster(internal way)

```

```
var cfg *rest.Config
var k8sClient client.Client
var ctx context.Context
var cancel context.CancelFunc
var testEnv *envtest.Environment

func TestAPIs(t *testing.T) {
	RegisterFailHandler(Fail)

	RunSpecsWithDefaultAndCustomReporters(t,
		"Controller Suite",
		[]Reporter{printer.NewlineReporter{}})
}

var _ = BeforeSuite(func() {
	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
	ctx, cancel = context.WithCancel(context.TODO())

	By("bootstrapping test environment")
	testEnv = &envtest.Environment{
		CRDDirectoryPaths:     []string{filepath.Join("..", "config", "crd", "bases")},
		ErrorIfCRDPathMissing: true,
		// https://github.com/operator-framework/operator-sdk/issues/5090#issuecomment-1001814557
		BinaryAssetsDirectory: "/Users/tifa/Library/Application Support/io.kubebuilder.envtest/k8s/1.22.1-darwin-amd64",
	}

	cfg, err := testEnv.Start()
	Expect(err).NotTo(HaveOccurred())
	Expect(cfg).NotTo(BeNil())

--omission--
```

In my situation, I need to use `cfg` directly, in the sub test cases.

So, if you guys think this is an issue, please fix this.

And I saw that some template may cover this, just like https://github.com/kubernetes-sigs/kubebuilder/blob/72f7c2ef3fd42d21756bc04029ecc499daee2533/pkg/plugins/golang/v2/scaffolds/internal/templates/controllers/controller_suitetest.go , and the others don't. 

And I can help on this issue, if you guys confirm that need to be fix.

**Comment by mogsie** [comment]
I think you're right.  The code generated by "create api" does seem to have a _nil_ cfg.

The simples thing is to replace the `:=` with `=` — but that requires that you declare the variable `err` up front:

Before:
```
cfg, err := testEnv.Start()
```

After:
```
var err error
cfg, err = testEnv.Start()
```

**Comment by camilamacedo86** [comment]
It seems sorted out in the latest versions
Could you please check if are able to face the same with a project built with the latest release?
if so, please open an issue for we better analyse your case.

I am closing this one
However, if you think that is required feel free to re-open,
